### PR TITLE
Support pack-to for fs type and add make_vdi participant

### PR DESCRIPTION
**Pull request by lbt**

### CODE REVIEW

File: src/img_boss/make_vdi.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        """
+        f = wid.fields
+
+        if not f.image or not f.image.image_type or not f.image.image_url:
+            raise RuntimeError("Missing mandatory field 'image'")
+
+        if not f.image.image_type == "raw":
+            return False, "Image type is not raw"
+
+        raw = os.path.abspath(f.image.image_url.replace(self.config.base_url,
```
Comment (keto) [inline src/img_boss/make_vdi.py]: Could this instead use

    os.path.join(f.image.outdir, os.path.basename(f.image.image_url))

so there wouldn't be need for configuration?

Though I'm not sure if outdir is intentionally included in the image data that ends up in the WI

File: src/img_boss/make_vdi.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            print " ".join(command)
+            subprocess.check_output(command)
+            subprocess.check_call(["chmod", "0664", vdi])
+            subprocess.check_call(["bzip2", "--fast", vdi])
+            subprocess.check_call(["chown", "-R", "img", os.path.dirname(raw)])
+        except subprocess.CalledProcessError,e:
+            return False, "Running convertfromraw failed:\n%s" % e.output
+
+        # Cleanup
+        os.remove(raw)
```
Comment (keto) [inline src/img_boss/make_vdi.py]: If the  raw image is removed, should the image_url, image_type and others that refer to it be updated it the WI?

File: src/img_boss/make_vdi.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                              raw)
+#                    new_raw = subprocess.check_output(["tar", "xv",
+#                                                       "-C", "/tmp",
+#                                                       "-f", raw]
+#                                                       ).rstrip('\n')
+#                    os.unlink(raw)
+#                    raw = raw[0:-8] + ".raw"
+#                    shutil.copy(os.path.join("/tmp", new_raw),
+#                                raw)
+#                    os.unlink(os.path.join("/tmp", new_raw))
```
Comment (keto) [inline src/img_boss/make_vdi.py]: Commented out code should be removed

File: src/img_boss/make_vdi.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        pass
+
+    def handle_lifecycle_control(self, ctrl):
+        """ participant control thread """
+        if ctrl.message == "start":
+            if ctrl.config.has_section("make_vdi"):
+                self.config = dap({})
+                self.config.base_url = ctrl.config.get("make_vdi", "base_url")
+                self.config.base_dir = ctrl.config.get("make_vdi", "base_dir")
+                print "config base_url %s" % self.config.base_url
```
Comment (keto) [inline src/img_boss/make_vdi.py]: Would be better to use self.log.debug/info/error instead of prints

File: src/img_boss/make_vdi.py (hunk 3 above)
Comment (lbt) [reply src/img_boss/make_vdi.py]: +1

File: src/img_boss/make_vdi.py (hunk 2 above)
Comment (lbt) [reply src/img_boss/make_vdi.py]: that makes sense

File: src/img_boss/make_vdi.py (hunk 1 above)
Comment (lbt) [reply src/img_boss/make_vdi.py]: I think there's some path data too - like web/lbt/ ??

Comment (lbt) [reply src/img_boss/make_vdi.py]: it seems that f.image.outdir is not set and the outdir isn't in fields so :(

File: src/img_boss/make_vdi.py (hunk 2 above)
Comment (keto) [reply src/img_boss/make_vdi.py]: I was thinking more like changing the image_type to 'vdi' and image_url to point to the vdi file

**Pull request merged**